locator/tablets.hh

```diff
@@ -202,6 +203,33 @@ enum tablet_range_side {
     right = 1,
 };
 
+// The decision of whether tablets of a given should be split, merged, or none, is made
+// by the load balancer. This decision is recorded in the tablet_map and stored in group0.
+struct resize_decision {
+    struct none {};
+    struct split {};
+    struct merge {};
+
+    std::variant<none, split, merge> way;
+    // Generation globally identifies a resize decision.
+    // It's monotonically increasing, globally (comes from group0_guard timestamp).
+    // Needed to distinguish stale decision from latest one, in case coordinator
+    // revokes a resize decision and signal it again later.
+    api::timestamp_type generation = api::min_timestamp;
+
+    resize_decision() = default;
+    resize_decision(sstring decision, api::timestamp_type);
+    operator bool() const {
+        return !std::holds_alternative<resize_decision::none>(way);
+    }
+    bool operator==(const resize_decision&) const;
+    // Generation determines the precedence of resize decision. A decision that came
+    // later in time will have higher precedence over the older one.
+    std::strong_ordering compare_by_generation(const resize_decision& other) const;
+    sstring name() const;
```

Comment:
What's the name of a resize decision?

Reply:
what do you want? a better name or doc?

Reply:
if we think resize decisions can be only one of merge, split or none. then name() as returning string form of the decision type, makes sense, no?

maybe type_name() ?

Reply:
Maybe just create a formatter for it and then use fmt::format()